T24025 Add setter support to the library

GeorgesStavracas opened the pull request:
https://phabricator.endlessm.com/T24025

(This is a clone of https://github.com/endlessm/eos-parental-controls/pull/6 for it shouldn't have been merged)

pwithnall committed:
libeos-parental-controls: Clarify ownership of a struct member

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T24025

pwithnall committed:
libeos-parental-controls: Add a missing (type) annotation

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T24025

pwithnall committed:
libeos-parental-controls: Add setter support for app filters

This includes some basic tests. Full test coverage has not yet been
achieved.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T24025

pwithnall committed:
eos-parental-controls-client: Add a ‘set’ command

This allows the app filter to be set, using the new library API for it.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T24025

pwithnall commented:
I’ve responded to the review comments from #6 and #7 (which is where Will did his review for this code) and have updated the branch, and added some new tests.

When this is merged, the new non-fixup commit should be moved to the start of the branch.

pwithnall committed:
libeos-parental-controls: Rename a callback function

Make it a bit clearer that it gets the bus purely for the getter method
for parental controls, which it is hard-coded to chain to.

This will clarify some later changes.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T24025

pwithnall commented:
Rebased, squashed and merged. Thanks.
